.github/actions/terraform-init/action.yml

```diff
@@ -38,21 +38,19 @@ runs:
         fi
       shell: bash
       
-    - uses: hashicorp/setup-terraform@v3
+    - uses: opentofu/setup-opentofu@v1
       with:
-        terraform_version: 1.5.7
+        terraform_version: 1.9.0
     
     - run: |
         cd ${{ inputs.dir }}
         backend_key=${{ inputs.backend-s3-key }}
         if [ -z "$backend_key" ]; then
           backend_key="aws/app/${{ inputs.env }}/${{ inputs.service-name }}/infrastructure.tfstate"
         fi
-        terraform init -reconfigure \
+        tofu init -reconfigure \
           -backend-config="bucket=fgr-terraform-state" \
-          -backend-config="dynamodb_table=fgr-terraform-state-locks" \
```

Comment:
Mně ten zámek hrozně točí. 😅 Strašně často se mi stává že nesedí checksum, imho v tý starší verzi terraformu co používáme je nějaký bug. Na druhou stranu v OpenTofu 1.10 který asi vyjde celkem brzy bude native zamykání v S3 takže to Dynamo nebude potřeba. (https://opentofu.org/blog/help-us-test-opentofu-1-10-0-alpha1/#native-locking-support-for-s3-backend) Tak bych ho pro teď odstranil a potom použil to s3.